hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestTableSnapshotScanner.java

```diff
@@ -244,6 +245,68 @@ public void testWithOfflineHBaseMultiRegion() throws Exception {
     testScanner(UTIL, "testWithMultiRegion", 20, true);
   }
 
+  @Test
+  public void testScanMetricsByRegionForTableSnapshotScanner() throws Exception {
+    TableName tableName = TableName.valueOf(name.getMethodName() + "_TABLE");
+    String snapshotName = name.getMethodName() + "_SNAPSHOT";
+    try {
+      createTableAndSnapshot(UTIL, tableName, snapshotName, 50);
+      Path restoreDir = UTIL.getDataTestDirOnTestFS(snapshotName);
+      Scan scan = new Scan().withStartRow(bbb).withStopRow(yyy); // limit the scan
+      scan.setScanMetricsEnabled(true);
+      scan.setEnableScanMetricsByRegion(true);
+      Scan scanCopy = new Scan(scan);
+
+      Configuration conf = UTIL.getConfiguration();
+
+      // Scan table snapshot and get combined scan metrics
```

Comment:
nit: each of these little blocks could be its own test method, and then this test would be a lot nicer to reason about when there are failures.